test/suite/test_tiered08.py

```diff
@@ -0,0 +1,124 @@
+# software under copyright law.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+# [TEST_TAGS]
+# checkpoints:correctness:checkpoint_data
+# [END_TAGS]
+#
+
+import os, threading, time, wiredtiger, wttest
+from wtthread import checkpoint_thread, flush_tier_thread
+
+# test_tiered08.py
+#   Run background checkpoings and flush_tier operations while inserting
+#   data into a table from another thread.
+class test_tiered08(wttest.WiredTigerTestCase):
+
+    nkeys = 200000
+    ckpt_freq = nkeys / 12
+    flush_freq = nkeys / 10
+
+    uri = "table:test_tiered08"
+
+    auth_token = "test_token"
+    bucket = "mybucket"
+    bucket_prefix = "pfx_"
+    extension_name = "local_store"
+    prefix = "pfx-"
+
+    def conn_config(self):
+        if not os.path.exists(self.bucket):
+            os.mkdir(self.bucket)
+        return \
+          'statistics=(all),' + \
+          'tiered_storage=(auth_token=%s,' % self.auth_token + \
+          'bucket=%s,' % self.bucket + \
+          'bucket_prefix=%s,' % self.prefix + \
+          'name=%s),tiered_manager=(wait=0)' % self.extension_name
+
+    # Load the local store extension, but skip the test if it is missing.
+    def conn_extensions(self, extlist):
+        extlist.skip_if_missing = True
+        extlist.extension('storage_sources', self.extension_name)
+
+    def key_gen(self, i):
+        return 'KEY' + str(i)
+
+    def value_gen(self, i):
+        return 'VALUE_' + 'filler' * (i % 12) + str(i)
+
+    def populate(self):
+        self.pr('Populating tiered table')
+        c = self.session.open_cursor(self.uri, None, None)
+        for i in range(self.nkeys):
+            c[self.key_gen(i)] = self.value_gen(i)
+        c.close()
+
+    def verify(self):
+        self.pr('Verifying tiered table')
+        c = self.session.open_cursor(self.uri, None, None)
+        for i in range(self.nkeys):
+            self.assertEqual(c[self.key_gen(i)], self.value_gen(i))
+        c.close()
+
+    def test_tiered08(self):
+        cfg = self.conn_config()
+        self.pr('Config is: ' + cfg)
+        intl_page = 'internal_page_max=16K'
+        base_create = 'key_format=S,value_format=S,' + intl_page
+        self.session.create(self.uri, base_create)
+
+        done = threading.Event()
+        ckpt = checkpoint_thread(self.conn, done)
+        flush = flush_tier_thread(self.conn, done)
+
+        # Start background threads and give them a chance to start
+        ckpt.start()
+        flush.start()
+        time.sleep(0.5)
+
+        self.populate()
+
+        done.set()
```

Comment:
Before this, do you want to add a loop and verify that at least N checkpoint and flush tier calls took place?

Reply:
I've extended the test to do this.

Reply:
I see you added `statistics=(all)` (although I think `statistics=(fast)` is probably sufficient). But I'm not seeing where the test is extended to check the stats. Can you point me to that?

Reply:
Ah.  I knew there must be another way to do this.  I didn't use the statistics.  Instead I pulled this information from the metadata. `checkpoint_count()` figures out how many checkpoints we did by looking at the number on the checkpoint name in the test table's metadata. `flush_count()` gets the number of flushes by extracting the object ID from the name of the `file:` piece of the table.

Probably better to use the statistics, since they actually count what we care about.  I'll code that up tomorrow.

Reply:
I was scanning the code for statistics and totally missed the existence of those two functions. There is definitely a stat for number of `flush_tier` calls. I don't see an obvious one for number of checkpoints.